src/main/mw.c

```diff
@@ -686,6 +690,110 @@ void processRx(void)
 
 }
 
+void getArmingChannel(modeActivationCondition_t *modeActivationConditions, uint8_t *armingChannel) {
+    for (int index = 0; index < MAX_MODE_ACTIVATION_CONDITION_COUNT; index++) {
+        modeActivationCondition_t *modeActivationCondition = &modeActivationConditions[index];
+    	if (modeActivationCondition->modeId) {
```

Comment:
I tested this actually and debugged and it works. it always finds the
arming channel and only filters if its there.

I will review once more as I already forgot about why it worked.
Op 15 jul. 2015 17:49 schreef "Petr Ledvina" <notifications@github.com>:

> In src/main/mw.c
> <https://github.com/cleanflight/cleanflight/pull/1128#discussion_r34693352>
> :
>
> > @@ -686,6 +690,110 @@ void processRx(void)
> >
> >  }
> >
> > +void getArmingChannel(modeActivationCondition_t *modeActivationConditions, uint8_t *armingChannel) {
> > +    for (int index = 0; index < MAX_MODE_ACTIVATION_CONDITION_COUNT; index++) {
> > +        modeActivationCondition_t *modeActivationCondition = &modeActivationConditions[index];
> > +    	if (modeActivationCondition->modeId) {
>
> This will find last non-zero modeId (and BOXARM is zero ... )
>
> maybe : modeActivationCondition->modeId == BOXARM ?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/cleanflight/cleanflight/pull/1128/files#r34693352>.
>